test/functional/autocmd/textyankpost_spec.lua

```diff
@@ -0,0 +1,166 @@
+local helpers = require('test.functional.helpers')
+local clear, eval, eq, insert = helpers.clear, helpers.eval, helpers.eq, helpers.insert
+local feed, execute, expect = helpers.feed, helpers.execute, helpers.expect
+
+
+describe('TextYankPost', function()
+  before_each(function()
+    clear()
+
+    -- emulate the clipboard so system clipboard isn't affected
+    execute('let &rtp = "test/functional/fixtures,".&rtp')
+
+    execute('let g:count = 0')
+    execute('autocmd TextYankPost * let g:event = copy(v:event)')
```

Comment:
If neovim starts to clear keys tests without `copy()` will fail. Clearing keys means avoiding two alloc/free pairs per event which is good. I do not see `dict_clear()` function though (BTW, Python authors thought this functionality is needed, VimL has only `filter(dict, 0)`).